src/components/ui/ConditionalButton.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * This program has been developed by students from the bachelor Computer Science at Utrecht University within the Software Project course.
+ * © Copyright Utrecht University (Department of Information and Computing Sciences)
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { Button, ButtonProps } from '@/src/components/ui/Button';
+import { useWeb3Modal } from '@web3modal/react';
+import { HiOutlineExclamationCircle } from 'react-icons/hi2';
+
+export type Condition = {
+  enabled: boolean;
```

Comment:
I would name this something like "when"
Because "enabled" is confusing when you're looking at a call to the ConditionalButton component with the disabled prop on the ConditionalButton ifykwim

"when" makes sense to me, because you show the associated "content" **when** the condition is true